core/auth/local/views.py

```diff
@@ -40,7 +40,7 @@ def logout():
 def new_user():
     username = request.form.get("username")
     password = request.form.get("password")
-    if current_user.has_role('admin') and current_user.is_active:
+    if type(current_user) != AnonymousUserMixin and current_user.has_role('admin') and current_user.is_active:
```

Comment:
Type checks should use `isinstance()`, or `is` / `is not` operators (for strict type checks). In this case, I would suggest:
```suggestion
    if not isinstance(current_user, AnonymousUserMixin) and current_user.has_role('admin') and current_user.is_active:
```
See [this SO answer](https://stackoverflow.com/a/152596/3223422) for more.